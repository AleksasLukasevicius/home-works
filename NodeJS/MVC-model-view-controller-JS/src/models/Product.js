export class Product {
  title;
  price;
  isAvailable;

  constructor(initTitle, initPrice, initIsAvailable) {
    (this.title = initTitle),
      (this.price = initPrice),
      (this.isAvailable = initIsAvailable);
  }
}
