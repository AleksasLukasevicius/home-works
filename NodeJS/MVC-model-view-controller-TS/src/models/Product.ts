export class Product {
  price: number;
  title: string;
  isAvailable: boolean;

  constructor(price: number, title: string, isAvailable: boolean) {
    this.price = price;
    this.title = title;
    this.isAvailable = isAvailable;
  }
}
