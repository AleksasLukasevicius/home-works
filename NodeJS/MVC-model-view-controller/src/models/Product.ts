export class Product {
  price: any;
  title: any;
  isAvailable: any;

  constructor(price: number, title: string, isAvailable: boolean) {
    this.price = price;
    this.title = title;
    this.isAvailable = isAvailable;
  }
}
