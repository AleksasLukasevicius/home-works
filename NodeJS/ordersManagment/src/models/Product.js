import Joi from "joi";

const productSchema = Joi.object({
  id: Joi.number(),
  name: Joi.string().trim().required(),
  price: Joi.number().required(),
  isAvailable: Joi.boolean().required(),
  imageURL: Joi.string().trim().lowercase().required(),
});

export class Product {
  id;
  name;
  price;
  isAvailable;
  imageURL;

  constructor({ id, name, price, isAvailable, imageURL }) {
    const newProductData = { id, name, price, isAvailable, imageURL };
    const productValidationResult = productSchema.validate(newProductData);

    if (productValidationResult.error) {
      throw Error(productValidationResult.error);
    }

    this.id = id;
    this.name = name;
    this.price = price;
    this.isAvailable = isAvailable;
    this.imageURL = imageURL;
  }
}
