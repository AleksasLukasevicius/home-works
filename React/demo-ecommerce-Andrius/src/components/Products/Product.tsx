import { ProductContainer } from "../../styles/ProductContainer";
import { TransparentButton } from "../../styles/TransparentButton";
import type { TProduct } from "../../types";
import type { handleBuyClick } from "../../ProductsContext/types";

export const Product = ({
  product,
  isProductInCart,
  handleBuyClick,
}: {
  product: TProduct;
  isProductInCart: boolean;
  handleBuyClick: handleBuyClick;
}) => {
  return (
    <ProductContainer>
      <img src={product.image} alt={`Product ${product.title}`} />
      <p>{product.title}</p>
      <p>{product.price}</p>

      <TransparentButton onClick={() => handleBuyClick(product.id)}>
        {isProductInCart ? "+" : "Buy"}
      </TransparentButton>

      {isProductInCart ? (
        <TransparentButton onClick={() => handleBuyClick(product.id)}>
          -
        </TransparentButton>
      ) : null}
    </ProductContainer>
  );
};
