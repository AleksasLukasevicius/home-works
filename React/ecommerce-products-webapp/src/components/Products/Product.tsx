import { ProductsContext } from "..";
import { type FC, useContext } from "react";
import type { TProductProps } from "./types";
import Button from "@mui/material/Button";

// export const Product = ({ product }: TProductProps) => {
export const Product: FC<TProductProps> = ({ product }) => {
  const { dispatch } = useContext(ProductsContext);

  return (
    <div className="product-card">
      <h3>{product.title}</h3>
      <p>{product.description}</p>
      <p>Price: {product.price} â‚¬</p>

      <Button
        variant="outlined"
        size="small"
        onClick={() =>
          dispatch({ type: "addProduct", payload: { productId: product.id } })
        }
      >
        Add to cart
      </Button>
      <Button
        variant="outlined"
        size="small"
        onClick={() =>
          dispatch({
            type: "deleteProduct",
            payload: { productId: product.id },
          })
        }
      >
        Delete from Cart
      </Button>
    </div>
  );
};
