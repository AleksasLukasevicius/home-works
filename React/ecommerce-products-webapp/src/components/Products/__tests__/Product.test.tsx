import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Product } from "../Product";

describe("Product", () => {
  it("should render Product", () => {
    render(<Product />);
    expect(screen.getByLabelText("product")).toBeVisible();
  });

  it("should show product title", () => {
    render(<Product />);
    expect(screen.getByText("Fjallraven", { exact: false }));
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Product />);

    expect(tree).toMatchSnapshot();
  });
});
