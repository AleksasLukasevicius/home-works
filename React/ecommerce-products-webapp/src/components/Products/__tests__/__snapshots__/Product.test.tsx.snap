// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product matches snapshot 1`] = `
<div
  aria-label="product"
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-m7q8m2-MuiPaper-root-MuiCard-root"
>
  <img
    alt="Sorry no image title"
    aria-label="product image"
    className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-1bevvzh-MuiCardMedia-root"
    height={100}
    src=""
  />
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Price: 
       â‚¬
    </p>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-uh4svx-MuiCardActions-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      +
    </button>
    <button
      aria-expanded={false}
      aria-label="show more"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-3ogxn6-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
